File: engine/engine-model/src/main/scala/server/maps/MapleDragon.scala
package server.maps

import java.awt.Point

class MapleDragon(private var _ownerId: Int, _position: Point, _stance: Int) extends AbstractAnimatedMapleMapObject {
  position_$eq(_position)
  stance_$eq(_stance)

  def ownerId: Int = _ownerId

  override def `type`(): MapleMapObjectType = MapleMapObjectType.DRAGON

  override def objectId: Int = _ownerId
}

File: engine/engine-model/src/main/java/server/maps/MapleDragon.java
package server.maps;

import java.awt.Point;

public class MapleDragon extends AbstractAnimatedMapleMapObject {
   private final Integer ownerId;

   public MapleDragon(Integer ownerId, Point position, Integer stance) {
      this.ownerId = ownerId;
      setObjectId(ownerId);
      setPosition(position);
      setStance(stance);
   }

   public Integer ownerId() {
      return ownerId;
   }

   @Override
   public MapleMapObjectType type() {
      return MapleMapObjectType.DRAGON;
   }
}
